x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,224 @@
+  EuiFieldPassword,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiSelect,
+} from '@elastic/eui';
+import type { NewPackagePolicy } from '@kbn/fleet-plugin/public';
+import { NewPackagePolicyInput } from '@kbn/fleet-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { RadioGroup } from './csp_boxed_radio_group';
+import { getPosturePolicy, NewPackagePolicyPostureInput } from './utils';
+
+type SetupFormatGCP = 'google_cloud_shell' | 'manual';
+const GCPSetupInfoContent = () => (
+  <>
+    <EuiSpacer size="l" />
+    <EuiTitle size="s">
+      <h2>
+        <FormattedMessage
+          id="xpack.csp.gcpIntegration.setupInfoContentTitle"
+          defaultMessage="Setup Access"
+        />
+      </h2>
+    </EuiTitle>
+    <EuiSpacer size="l" />
+    <EuiText color={'subdued'} size="s">
+      <FormattedMessage
+        id="xpack.csp.gcpIntegration.setupInfoContent"
+        defaultMessage="The integration will need elevated access to run some CIS benchmark rules. Select your preferred
+    method of providing the GCP credentials this integration will use. You can follow these
+    step-by-step instructions to generate the necessary credentials."
+      />
+    </EuiText>
+  </>
+);
+
+const DocsLink = (
+  <EuiText color={'subdued'} size="s">
+    <FormattedMessage
+      id="xpack.csp.eksIntegration.docsLink"
+      defaultMessage="Read the {docs} for more details"
+      values={{
+        docs: (
+          <EuiLink
+            href="https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html"
+            external
+          >
+            documentation
+          </EuiLink>
+        ),
+      }}
+    />
+  </EuiText>
+);
+
+interface GcpFields {
+  fields: Record<string, { label: string; type?: 'password' | 'text' }>;
+}
+const gcpField: GcpFields = {
+  fields: {
+    project_id: {
+      label: i18n.translate('xpack.csp.gcpIntegration.projectidLabel', {
+        defaultMessage: 'Project ID',
+      }),
+    },
+    credentials_file: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsLabel', {
+        defaultMessage: 'Credentials File',
+      }),
+    },
+    credentials_json: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsjsonLabel', {
+        defaultMessage: 'Credentials JSON',
+      }),
+    },
+  },
+};
+
+const jsonOptionsHolder = [
+  { label: 'Option A', text: 'Option A' },
+  { label: 'Option B', text: 'Option B' },
+];
+
+const getSetupFormatOptions = (): Array<{
+  id: SetupFormatGCP;
+  label: string;
+  disabled: boolean;
+}> => [
+  {
+    id: 'google_cloud_shell',
+    label: 'Google Cloud Shell',
+    disabled: true,
+  },
+  {
+    id: 'manual',
+    label: i18n.translate('xpack.csp.gcpIntegration.setupFormatOptions.manual', {
+      defaultMessage: 'Manual',
+    }),
+    disabled: false,
+  },
+];
+
+export type AwsCredentialsType = keyof typeof gcpField;
+
+interface Props {
```

Comment:
Thats a big file with a lot of types and interfaces, i suggest we give a more definitive name to this type